crates/lgn-api-codegen-node/src/lib.rs

```diff
@@ -58,3 +58,73 @@ pub fn generate(options: GenerateOption) -> Result<()> {
 
     Ok(())
 }
+
+#[derive(Debug)]
+#[napi(object)]
+pub struct ApiOption {
+    /// Path to the _folder_ that contains all the apis definition files (.yaml), prefer absolute paths
+    pub path: String,
+    /// Name(s) of the api to generate the client code for
+    pub names: Vec<String>,
+    /// Filename without prefix nor extension
+    pub filename: String,
+}
+
+#[derive(Debug)]
+#[napi(object)]
+pub struct GenerateAllOption {
+    /// Output directory, prefer absolute paths
+    pub out_dir: String,
+    /// Path to a Prettier config file
+    pub prettier_config_path: Option<String>,
+    /// Generates a full-fledged node module including a `package.json` file
+    pub with_package_json: Option<bool>,
+    /// Skips code formatting
+    pub skip_format: Option<bool>,
+    /// Aliases for external API references
+    pub alias_mappings: Option<HashMap<String, String>>,
+    /// Api options
+    pub api_options: Vec<ApiOption>,
+}
+
+/// Generate all api clients.
+///
+/// # Errors
+///
+/// Throws if the generation fails (file not found, invalid, etc...)
+#[allow(clippy::needless_pass_by_value)]
+#[napi]
+pub fn generate_all(options: GenerateAllOption) -> Result<()> {
+    let prettier_config_path = options.prettier_config_path.map(PathBuf::from);
+
+    let alias_mappings: TypeScriptAliasMappings = options
+        .alias_mappings
+        .unwrap_or_default()
+        .try_into()
+        .map_err(|err: lgn_api_codegen::errors::Error| Error::from_reason(err.to_string()))?;
+
+    let result = options
+        .api_options
+        .into_iter()
+        .map(|api| {
+            lgn_api_codegen::generate(
+                Language::TypeScript(TypeScriptOptions {
+                    alias_mappings: alias_mappings.clone(),
+                    filename: api.filename,
+                    with_package_json: options.with_package_json.unwrap_or_default(),
+                    skip_format: options.skip_format.unwrap_or_default(),
+                    prettier_config_path: prettier_config_path.clone(),
+                }),
+                &api.path,
+                api.names,
+                &options.out_dir,
+            )
+        })
+        .collect::<lgn_api_codegen::errors::Result<Vec<Vec<PathBuf>>, _>>();
```

Comment:
@ereOn we have this use case in TypeScript where we need to generate the API clients for all the .yaml files located in all the apis folders. Unfortunately it's not supported by the generator itself and it could involve bigger refactors. So what I did is to add a function here in `lgn-api-codegen-node` that iterates through all the provided set of api folders/api names and generate a .ts file for each iteration. The code also regenerates the index.ts file each time (but it's not a big deal).

Do you think there is a better way to handle this use case? Is there any plans to integrate it to the main codegen crate?